File: src/main/java/tasks/siriusPreparing/Dogs.py
import math

n = int(input())
a = list(map(int, input().split()))
b = list(map(int, input().split()))

res = 1
minNegatives = math.inf

for start in range(n):
    negatives = 0
    collarInd = 0
    for i in range(n):
        dogInd = (start + i) % n
        dogR = b[dogInd]

        collarR = a[collarInd]
        collarInd += 1
        if collarR > dogR:
            continue
        elif dogR - collarR <= 100:
            negatives += (dogR - collarR) // 2
        else:
            negatives += 30

    if negatives < minNegatives:
        res = start + 1
        minNegatives = negatives

print(res, minNegatives)

File: src/main/java/tasks/siriusPreparing/Dogs.java
package tasks.siriusPreparing;

import java.util.Scanner;

public class Dogs {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        int n = scanner.nextInt();
        int[] a = new int[n];
        int[] b = new int[n];

        for (int i = 0; i < n; i++) {
            a[i] = scanner.nextInt();
        }

        for (int i = 0; i < n; i++) {
            b[i] = scanner.nextInt();
        }

        int res = 0;
        long minNegatives = Long.MAX_VALUE;
        for (int start = 0; start < n; start++) {
            int negatives = 0;
            for (int collarInd = 0; collarInd < n; collarInd++) {
                int dogInd = (start + collarInd) % n;
                int dogR = b[dogInd];
                int collarR = a[collarInd];

                if (dogR > collarR) {
                    continue;
                } else if (collarR - dogR <= 100) {
                    negatives += (collarR - dogR) / 2;
                } else {
                    negatives += 30;
                }
            }
            if (negatives < minNegatives) {
                res = start + 1;
                minNegatives = negatives;
            }
        }
        System.out.println(res + " " + minNegatives);
    }
}

